content/en/docs/concepts/architecture/master-node-communication.md

```diff
@@ -97,13 +97,225 @@ public networks.
 
 ### SSH Tunnels
 
-Kubernetes supports SSH tunnels to protect the Master -> Cluster communication
+Kubernetes supports SSH tunnels to protect the Master → Cluster communication
 paths. In this configuration, the apiserver initiates an SSH tunnel to each node
 in the cluster (connecting to the ssh server listening on port 22) and passes
 all traffic destined for a kubelet, node, pod, or service through the tunnel.
 This tunnel ensures that the traffic is not exposed outside of the network in
 which the nodes are running.
 
-SSH tunnels are currently deprecated so you shouldn't opt to use them unless you know what you are doing. A replacement for this communication channel is being designed.
+SSH tunnels are currently deprecated so you shouldn't opt to use them unless you
+know what you are doing. The Konnectivity service is a replacement for this
+communication channel.
+
+### Konnectivity service
+{{< feature-state for_k8s_version="v1.18" state="beta" >}}
+
+As a replacement to the SSH tunnels, the Konnectivity service provides TCP
+level proxy for the Master → Cluster communication. The feature is beta
+in 1.18.
+
+To use it on GCE, you can deploy the cluster using kube-up.sh in one line:
+
+```bash
+KUBE_ENABLE_EGRESS_VIA_KONNECTIVITY_SERVICE=true ./cluster/kube-up.sh
+```
+
+Let's look at what kube-up.sh does, so that you can use the Konnectivity service
+in other environments with some modifications.
+
+First, you need to configure the API Server to use the Konnectivity service
+to direct its network traffic to cluster nodes:
+1. Set the `--egress-selector-config-file` flag of the API Server, it is the
+path to the API Server egress configuration file.
+2. At the path, create a configuration file. For example,
+
+```yaml
+apiVersion: apiserver.k8s.io/v1beta1
+kind: EgressSelectorConfiguration
+egressSelections:
+# Since we want to control the egress traffic to the cluster, we use the
+# "cluster" as the name. Other supported values are "etcd", and "master".
+- name: cluster
```

Comment:
We should include master and etcd and just note that they should be kept at their default values of direct connection. They "technically" need to be defined.

Reply:
I omitted them because this doc is about master->cluster traffic. And omitting "etcd" and "master" is ok, because by default they would do direct dial. e.g., [etcd dial](https://github.com/kubernetes/kubernetes/pull/82048/files#diff-d02b855276b48d62bd9ec183560db2e2R126).